### PR TITLE
Add test for revalidated a cached worker script from a COEP: require-…

### PR DESCRIPTION
**Pull request by cdumez**
…corp document

This is an upstream for WebKit's https://github.com/WebKit/WebKit/pull/5141.

### CODE REVIEW

File: html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    headers = []
+    inm = request.headers.get(b'if-none-match', None)
+    if inm:
+        status = 304, u"Not Modified"
+        return status, headers, u""
+    else:
+        headers.append((b"Content-Type", b"text/javascript"))
+        headers.append((b"Access-Control-Allow-Origin", b"*"))
+        headers.append((b"Cross-Origin-Embedder-Policy", b"require-corp"))
+        headers.append((b"Cross-Origin-Resource-Policy", b"cross-origin"))
```
Comment (annevk) [inline html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py:18]: It seems that this header can be dropped.

File: html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        return u''
+
+    headers = []
+    inm = request.headers.get(b'if-none-match', None)
+    if inm:
+        status = 304, u"Not Modified"
+        return status, headers, u""
+    else:
+        headers.append((b"Content-Type", b"text/javascript"))
+        headers.append((b"Access-Control-Allow-Origin", b"*"))
```
Comment (annevk) [inline html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py:16]: This header isn't needed either. Workers cannot be cross-origin.

File: html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+
+
+def main(request, response):
+    # CORS preflight
+    if request.method == u'OPTIONS':
+        return u''
```
Comment (annevk) [inline html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py:7]: This can also be dropped.

**Review by annevk — APPROVED** [review]
Seems good, but I'd recommend dropping the no-op bits to avoid confusion.

File: html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py (hunk 1 above)
Comment (cdumez) [reply html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py]: Fixed.

File: html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py (hunk 2 above)
Comment (cdumez) [reply html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py]: Fixed.

File: html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py (hunk 3 above)
Comment (cdumez) [reply html/cross-origin-embedder-policy/resources/dedicated-worker-supporting-revalidation.py]: Fixed.

**Pull request merged**